### PR TITLE
Add Media.prototype.setRate method (only for iOS)

### PR DESCRIPTION
**Pull request by whitecolor**
This is refered to https://issues.apache.org/jira/browse/CB-8098
Adds playback change rate feature (setRate) for iOS. To add it it Android and WP it seems not quite easy.

### CODE REVIEW

**Comment by purplecabbage** [comment]
I would argue that if "To add it it Android and WP it seems not quite easy." then it should NOT be added at all.

The way this is implemented currently, any call to setRate on platforms other that iOS will still reach native code and potentially crash.  At a minimum, I would wrap the exec call in Media.js : Media.prototype.setRate so that it is not called on other platforms, and simply outputs NOTSUPPORTED or something.

**Comment by whitecolor** [comment]
@purplecabbage ok I will add this. An importaint question: how this PR is going to be tested, and what actions should be done before it is merged into master and be included in official version?

**Comment by purplecabbage** [comment]
This API has not really been discussed, but I have thought about it in the past.
My feeling is that it would be more useful to add the playbackRate as an additional option to Media.play
This would have the benefit of not breaking any platforms, they would simply ignore the unknown option ... and could possibly implement it in the future.
This would however mean that you could NOT change the playback rate of a media object that was already playing.  Not sure if this was a requirement, but I think it is a reasonable trade-off as controlling playback rate of a playing media file would probably be glitchy from JS.

**Comment by whitecolor** [comment]
@purplecabbage Well I think the variant with playbackRate option for ```play``` maybe ok but lets consider this most of the platforms treats playback rate just like ```volume``` property, that can be changed at any time:

1) HTML5 audio supports just a property ```playbackRate``` on ```audio``` element, that you can set anytime while playing.

2) iOs supports ```setRate``` method on ```AVAudioPlayer``` object (to can be set while playing).(https://developer.apple.com/library/ios/documentation/AVFoundation/Reference/AVAudioPlayerClassReference/index.html)

3) Windows Phone supports ```SpeedRatio``` property on ```MediaElement``` (http://msdn.microsoft.com/en-us/library/system.windows.controls.mediaelement.speedratio(v=vs.110).aspx)

4) The problem with android: phonegap uses ```MediaPlayer``` for playing media files, but it does not support any rate change methods (an old issue https://code.google.com/p/android/issues/detail?id=1961), there is also ```AudioTrack``` class, that can be used to play media, and probably have some capabilities of chanding rate (need to be tested) but there is a problem with seeking and actually it plays only uncompressed audio (that can be taken from ```MediaPlayer``` - so the workflow with using ```AudioTrack``` is quite complicated). The is also ```SoundPool``` class that has method ```setRate```, but it has the same problems that ```AudioTrack``` class. But actually people do implement such capabilities in android (in different audiobooks playing applicatons).

So I think it is reasonable to approach to ```payback rate``` just as to ```volume``` property, and as PG has method ```setVolume``` and had proposed ```setRate``` for this. What do you think?

**Comment by purplecabbage** [comment]
Yeah, ```playbackRate``` is the right choice then.  Following the HTML5 audio element spec is more important than my alternative suggestion.

**Comment by whitecolor** [comment]
Do what would you sugget to do with ```playbackRate```? create method ```setPlaybackRate``` (insteald of ```setRate```) or what? in HTML audio's ```playbackRate``` actually JS getter/setter.

**Comment by purplecabbage** [comment]
For now just go back to ```Media.prototype.setRate = function(rate) { ...```
But add the check for ios, to prevent crashing on platforms that do not implement it.

**Comment by whitecolor** [comment]
Ok, I've added, if ok with this check, how it is going to be tested?

**Comment by whitecolor** [comment]
Any comments?

**Comment by shazron** [comment]
I think you added the check wrongly - you added it to setVolume?

As for tests, we have a bunch of auto tests written in jasmine: https://github.com/apache/cordova-plugin-media/tree/master/tests
.. that is run using cordova-mobile-spec (see the repo)

File: www/Media.js [hunk 1]
```diff
@@ -149,9 +149,21 @@ Media.prototype.release = function() {
  * Adjust the volume.
  */
 Media.prototype.setVolume = function(volume) {
-    exec(null, null, "Media", "setVolume", [this.id, volume]);
+    if (device.platform == 'iOS'){
```
Comment (purplecabbage) [inline www/Media.js]: Doesn't this require the device plugin to be installed?
Instead, you should use ``` if (cordova.platformId === 'ios') ...```

File: www/Media.js [hunk 2]
```diff
@@ -149,9 +149,21 @@ Media.prototype.release = function() {
...
+        exec(null, null, "Media", "setVolume", [this.id, volume]);
+    } else {
+        console.warn('media.setRate method is currently not supported for', device.platform, 'platform.')
```
Comment (purplecabbage) [inline www/Media.js]: and here ... cordova-plugin-device is not a dependency, and shouldn't be

**Comment by whitecolor** [comment]
Sorry, my bad, would you like me to do a rebase or fixing commit on top?

**Comment by whitecolor** [comment]
I've fixed it.

File: src/ios/CDVSound.m [hunk 3]
```diff
@@ -686,6 +714,7 @@ @implementation CDVAudioFile
 @synthesize resourcePath;
 @synthesize resourceURL;
 @synthesize player, volume;
+@synthesize player, rate;
```
Comment (spenceropope) [inline src/ios/CDVSound.m]: I'm getting a problem with line 717 due to "player" already being implemented in line 716.

Comment (whitecolor) [inline src/ios/CDVSound.m]: How the code should look like?
```objective-c
 @synthesize player, volume, rate;
```

?

**Comment by purplecabbage** [comment]
@synthesize player, volume, rate;
OR
@synthesize player, volume;
@synthesize rate;

**Comment by whitecolor** [comment]
Fixed it.

**Comment by whitecolor** [comment]
Hey guys what about this pull is going to be merged? Are there any problems with it?

**Comment by whitecolor** [comment]
Any changes on this?

**Comment by whitecolor** [comment]
Great. When it is going to be available via PG build?

**Comment by whitecolor** [comment]
Is there a know date when the new version is going to be published?

**Pull request merged**